sql: add execution stats to node_statement_statistics

asubiotto opened the pull request:
Release justification: low-risk, high-benefit change to existing functionality
since this commit exposes pre-existing execution stats through the SQL shell.

Release note (sql change): sampled execution stats are now available through
node_statement_statistics.

Closes #61637 

Putting @yuzefovich as primary reviewer but @maryliag and @Azhng, please take a look and let me know if you have any questions.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/62006)
<!-- Reviewable:end -->

asubiotto commented:
I did not run any tests locally, so I'm expecting CI to be red. Will fix and repush once that happens.

asubiotto commented:
I'll also update `node_transaction_statistics`. You can hold off reviewing until I update this PR.

asubiotto commented:
OK, should be ready for a review now.

yuzefovich reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

A couple of nits which you should feel free to ignore.

> Reviewed 4 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/62006#-:-MVqsKGP3xgA_j7CTzej:bcx6fe5)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @asubiotto, @Azhng, and @maryliag)

___
*[pkg/sql/crdb_internal.go, line 861 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/62006#-MVqrEZ354cA2VZvvE1I:-MVqrEZ475J_ji0B_GjR:b-qa4hdv) ([raw file](https://github.com/cockroachdb/cockroach/blob/8fb0a6ff70708afbccb1b1f4280fd40a46770cf4/pkg/sql/crdb_internal.go#L861)):*
> ```Go
>   rows_read_avg       FLOAT NOT NULL,
>   rows_read_var       FLOAT NOT NULL,
>   network_bytes_avg   FLOAT,
> ```

super nit: it might be nicer to put two "network" fields next each other as well as two "memory usage" fields.
___
*[pkg/sql/crdb_internal.go, line 964 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/62006#-MVqrwQJDSLGI8CeU69T:-MVqrwQJDSLGI8CeU69U:b-utjeh0) ([raw file](https://github.com/cockroachdb/cockroach/blob/8fb0a6ff70708afbccb1b1f4280fd40a46770cf4/pkg/sql/crdb_internal.go#L964)):*
> ```Go
> 					tree.NewDFloat(tree.DFloat(s.mu.data.RowsRead.Mean)),
> 					tree.NewDFloat(tree.DFloat(s.mu.data.RowsRead.GetVariance(s.mu.data.Count))),
> 					execStatAvg(s.mu.data.ExecStats.Count, s.mu.data.ExecStats.NetworkBytes),
> ```

super nit: it might be helpful to put an inlined comment for all of these with the column name, e.g. this line would be `// network_bytes_avg`. We seem to follow such pattern for some virtual tables, and for this one in particular it would beneficial since it is very wide.


<!-- Sent from Reviewable.io -->

asubiotto reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/62006)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @Azhng, @maryliag, and @yuzefovich)

___
*[pkg/sql/crdb_internal.go, line 861 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/62006#-MVqrEZ354cA2VZvvE1I:-MVrb3id9KarX6uZi6YD:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/8fb0a6ff70708afbccb1b1f4280fd40a46770cf4/pkg/sql/crdb_internal.go#L861)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

super nit: it might be nicer to put two "network" fields next each other as well as two "memory usage" fields.
</blockquote></details>

Done.
___
*[pkg/sql/crdb_internal.go, line 964 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/62006#-MVqrwQJDSLGI8CeU69T:-MVrb4QoBf2Z1I_tyk41:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/8fb0a6ff70708afbccb1b1f4280fd40a46770cf4/pkg/sql/crdb_internal.go#L964)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

super nit: it might be helpful to put an inlined comment for all of these with the column name, e.g. this line would be `// network_bytes_avg`. We seem to follow such pattern for some virtual tables, and for this one in particular it would beneficial since it is very wide.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

yuzefovich reviewed (approved):
Thanks!

> Reviewed 4 of 4 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/62006#-:-MVrvdU2D8bghZI91X9s:b-ucfknh)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @asubiotto, @Azhng, and @maryliag)

___
*[pkg/sql/crdb_internal.go, line 869 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/62006#-MVrvJnlExTy8AVHu4FM:-MVrvJnlExTy8AVHu4FN:bgcapxs) ([raw file](https://github.com/cockroachdb/cockroach/blob/23567e996bb49ddc29b0d520300cc5ba507f027b/pkg/sql/crdb_internal.go#L869)):*
> ```Go
>   max_disk_usage_avg  FLOAT,
>   max_disk_usage_var  FLOAT,
>   contention_time_avg FLOAT, 
> ```

super nit: seems like a dangling space.


<!-- Sent from Reviewable.io -->

Commit:
sql: add execution stats to node_{statement,transaction}_statistics

Release justification: low-risk, high-benefit change to existing functionality
since this commit exposes pre-existing execution stats through the SQL shell.

Release note (sql change): sampled execution stats are now available through
node_{statement,transaction}_statistics.

asubiotto reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/62006)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @Azhng, @maryliag, and @yuzefovich)

___
*[pkg/sql/crdb_internal.go, line 869 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/62006#-MVrvJnlExTy8AVHu4FM:-MVrw83_2-FxFl_IwCc8:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/23567e996bb49ddc29b0d520300cc5ba507f027b/pkg/sql/crdb_internal.go#L869)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

super nit: seems like a dangling space.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

asubiotto commented:
TFTR

bors r=yuzefovich
